### PR TITLE
Project skeleton, general interfaces

### PR DESCRIPTION
**Pull request by DanielChanger**
**Description**:
Added project structure, basic user API and Bring abstractions.

**Ticket**: 
https://trello.com/c/4g070sGh

### CODE REVIEW

File: src/main/java/com/bobocode/hoverla/bring/annotation/Bean.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation which is used to mark types as beans in order to be scanned by {@link ApplicationContext}
+ */
+@Target(ElementType.TYPE)
```
Comment (nourish1709) [inline src/main/java/com/bobocode/hoverla/bring/annotation/Bean.java:13]: How about add `ElementType.METHOD` here? For Java Configuration

File: src/main/java/com/bobocode/hoverla/bring/context/BeanFactory.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    /**
+     * Returns a bean instance by its name and type.
+     *
+     * @param beanName name of the desired instance
+     * @param beanType type of the desired instance
+     * @param <T>      type of the desired instance
+     * @return object of type T which represents desired instance
+     * @throws NoSuchBeanException if instance with provided name and type is not found
+     */
+    <T> T getBean(String beanName, Class<T> beanType);
```
Comment (nourish1709) [inline src/main/java/com/bobocode/hoverla/bring/context/BeanFactory.java:43]: May you explain please, why is this method needed? All beanNames are unique as I understand, so beanType here adds no qualifying

**Review by nourish1709 — CHANGES_REQUESTED** [review]

File: src/main/java/com/bobocode/hoverla/bring/context/BeanFactory.java (hunk 2 above)
Comment (NazarSmith228) [reply src/main/java/com/bobocode/hoverla/bring/context/BeanFactory.java:43]: @nourish1709 What if we have 3 different beans of the same type (interface for example) ? Beans are unique by name, not by type.

Comment (nourish1709) [reply src/main/java/com/bobocode/hoverla/bring/context/BeanFactory.java:43]: > @nourish1709 What if we have 3 different beans of the same type (interface for example) ? Beans are unique by name, not by type.

as I understand, bean type just adds type safety, doesn't it?

Comment (NazarSmith228) [reply src/main/java/com/bobocode/hoverla/bring/context/BeanFactory.java:43]: @nourish1709 exactly. It is similar to `getBean(String name)`, but it is type safe.

File: src/main/java/com/bobocode/hoverla/bring/annotation/Bean.java (hunk 1 above)
Comment (DanielChanger) [reply src/main/java/com/bobocode/hoverla/bring/annotation/Bean.java:13]: I think it will be a part some of the stories related to java configuration

**Review by nourish1709 — APPROVED** [review]

**Review by nadya-gainutsa — APPROVED** [review]

**Pull request merged**